src/dlload.c

```diff
@@ -183,13 +184,21 @@ JL_DLLEXPORT void *jl_load_dynamic_library(const char *modname, unsigned flags,
                 size_t len = strlen(dl_path);
                 if (len == 0)
                     continue;
+
+                // Is this entry supposed to be relative to the bindir?
+                if (strncmp(dl_path, "@executable_path", 16 < len ? 16 : len) == 0) {
```

Comment:
```suggestion
                if (len > 17 && strcmp(dl_path, "@executable_path/") == 0) {
```

Reply:
Wait, I misread this.  This changes it to only match exactly `"@executable_path/"`; I want it to match all strings beginning with `"@executable_path/"`.

Reply:
You probably still want the `len > 16`?

Reply:
Sure, let's save a couple of cycles.  Even without it, it should still function correctly, but it will be faster than string comparing those first couple of characters. :)